addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/device/BulbDevice.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.tplinksmarthome.internal.device;
+
+import static org.openhab.binding.tplinksmarthome.TPLinkSmartHomeBindingConstants.*;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.tplinksmarthome.internal.Commands;
+import org.openhab.binding.tplinksmarthome.internal.Connection;
+import org.openhab.binding.tplinksmarthome.internal.TPLinkSmartHomeConfiguration;
+import org.openhab.binding.tplinksmarthome.internal.model.HasErrorResponse;
+import org.openhab.binding.tplinksmarthome.internal.model.LightState;
+import org.openhab.binding.tplinksmarthome.internal.model.TransitionLightStateResponse;
+
+/**
+ * TP-Link Smart Home Light Bulb.
+ *
+ * @author Hilbrand Bouwkamp - Initial contribution
+ */
+@NonNullByDefault
+public class BulbDevice extends SmartHomeDevice {
+
+    protected Commands commands = new Commands();
+
+    private final int[] colorTempMinMax;
+    private final int colorTempRangeFactor;
+
+    public BulbDevice(ThingTypeUID thingTypeUID) {
+        if (THING_TYPE_LB130.equals(thingTypeUID)) {
+            colorTempMinMax = COLOR_TEMPERATURE_LB130_MIN_MAX;
+        } else {
+            colorTempMinMax = COLOR_TEMPERATURE_LB120_MIN_MAX;
+        }
+        colorTempRangeFactor = (colorTempMinMax[1] - colorTempMinMax[0]) / 100;
+    }
+
+    @Override
+    public String getUpdateCommand() {
+        return Commands.getRealtimeBulbAndSysinfo();
+    }
+
+    @Override
+    public boolean handleCommand(String channelID, Connection connection, Command command,
+            TPLinkSmartHomeConfiguration configuration) throws IOException {
+        int transitionPeriod = configuration.transitionPeriod;
+        final HasErrorResponse response;
```

Comment:
Can you leave out `final` for local variables, on all occurrences. The added value does not weight up to the added fuzz and if not used consistently it will not aid other contributors.